zlib-rs/src/deflate.rs

```diff
@@ -1358,6 +1355,10 @@ enum DataType {
 impl<'a> State<'a> {
     pub const BIT_BUF_SIZE: u8 = BitWriter::BIT_BUF_SIZE;
 
+    pub(crate) fn w_bits(&self) -> usize {
```

Comment:
it should not be needed, but you could try making this a `const fn`, and/or add a `#[inline(always)]` annotation. I don't expect either to do anything (this function is tiny so should always be inlined anyway), but maybe it has some unforseen effect.